docs/examples/musl_library.md

```diff
@@ -0,0 +1,250 @@
+# Comparing library
+
+In this example, we will demonstrate how to use DiffKemp on more complex
+programs like libraries. This can be useful for checking if some
+changes/refactoring in a library's codebase didn't cause semantic differences
+in the library API. We will use the [musl libc](https://musl.libc.org/) library
+and we will learn how to:
+
+- save the result of the comparison to a directory,
+- interpret the content of the output directory and
+- use the result viewer for visualisation of the differences.
+
+## 1. Getting the source files
+
+We will compare two versions of the library, the chosen versions are `1.2.3`
+and `1.2.4` but you can try different versions or even different libraries.
+First, we need to get the source files, we can get them from
+[here](https://musl.libc.org/releases.html) or we can download and extract
+them with the following commands:
+
+```sh
+wget -qO- https://musl.libc.org/releases/musl-1.2.3.tar.gz | tar xz
+wget -qO- https://musl.libc.org/releases/musl-1.2.4.tar.gz | tar xz
+```
+
+After downloading them and extracting them from the archive we need to
+configure them. We can do this with the following commands:
+
+```sh
+(cd musl-1.2.3/ && ./configure)
+(cd musl-1.2.4/ && ./configure)
+```
+
+## 2. Creating snapshots
+
+Similar to [the simple program example](simple_program.md), we first need to
+create snapshots of the program versions. We will save the snapshots as
+`snapshot-musl-1.2.3` and `snapshot-musl-1.2.4`. We could make the snapshots
+with the following command:
+
+```sh
+diffkemp build musl-1.2.3 snapshot-musl-1.2.3
+diffkemp build musl-1.2.4 snapshot-musl-1.2.4
+```
+
+Because the `compare` command by default compares all functions, there would be
+a lot of functions. We can change this by specifying a list of functions
+which should be analyzed. This can be useful if you want to try it on your
+library which has API functions for which you want to check if the semantic did
+not change without checking all functions.
+
+We can specify the functions by creating a file with the function
+names (each name on a single line). Let's name the file `function_list`. It
+could look for example like this:
+
+```txt
+sin
+cos
+open
+sem_post
+dlopen
+fork
+```
+
+Then to build the snapshots we will provide the command with the location of the
+file:
+
+```sh
+diffkemp build musl-1.2.3 snapshot-musl-1.2.3 function_list
+diffkemp build musl-1.2.4 snapshot-musl-1.2.4 function_list
+```
+
+Building of the snapshots will take some time.
+
+## 3. Comparing two versions of the library
+
+After creating the snapshots, we can compare them. Because there are many
+functions, using the `--stdout` argument (printing the result to standard
```

Comment:
Maybe we could mention `--report-stat` here since that allows to see at least some statistics about the analysis that could be useful?

Reply:
Thanks, I added it to the example and added mention about the `--extended-stat` option.